unittests/token_test_main.c

```diff
@@ -38,8 +38,49 @@ void test_token(void)
     ts_close(fd);
 }
 
+void test_token_utf8(void)
+{
+    cst_tokenstream *fd;
+    const char *token;
+    const char *singl = "€(){}[]";    /* EURO SIGN + defaults */
+    const char *prepunc = "¿¡\"!";    /* INVERTED QUESTION MARK, INV. EXCL. MARK, ", ! */
+
+    fd = ts_open("data_utf8.txt", " \n\t", singl, prepunc, NULL);       /* defaults */
+
+    token = ts_get(fd);
+    TEST_CHECK(strcmp(token, "Hello") == 0);
+    token = ts_get(fd);
+    TEST_CHECK(strcmp(token, "world") == 0);
+    TEST_CHECK(strcmp(fd->postpunctuation, ".") == 0);
+    token = ts_get(fd);
+    TEST_CHECK(strcmp(token, "Yahoo") == 0);
+    TEST_CHECK(strcmp(fd->postpunctuation, "!\"") == 0);
+    token = ts_get(fd);
+    TEST_CHECK(strcmp(token, "who") == 0);
+    token = ts_get(fd);
+    TEST_CHECK(strcmp(token, "(") == 0);
+    token = ts_get(fd);
+    TEST_CHECK(strcmp(token, "s") == 0);
+    token = ts_get(fd);
+    TEST_CHECK(strcmp(token, ")") == 0);
+    token = ts_get(fd);
+    TEST_CHECK(strcmp(token, "Good") == 0);
+    TEST_CHECK(strcmp(fd->prepunctuation, "¡") == 0);
+    token = ts_get(fd);
+    TEST_CHECK(strcmp(token, "News") == 0);
```

Comment:
I have improved the code coverage to make sure that all character sizes are used (I even used an smiley). I also checked that the actual token could contain special UTF-8 characters.  :smile: